Add Overview Custom Node with Trainer checkbox

lancelotblanchard opened the pull request:
Now, Editors can set their own `customNodeType` parameter (defaults to `Common`) to another Custom Node class. That's now the case with the Overview editor which has an `OverviewCustom` node, with a `Trainer` checkbox.

gidonkatten reviewed (changes requested):
One bug I noticed is what happens if you edit a function and delete `return` or add a `return`. Then the change is not reflected in the nodes. I have a fix but it would probably be best if we discussed as I am unsure exactly what we need

lancelotblanchard commented:
Good catch! I fixed this using a new  `returnStatement: { removed: boolean; added: boolean };` parameter in the `FuncDiff` interface. Let me know if that looks good to you.

gidonkatten commented:
> Good catch! I fixed this using a new `returnStatement: { removed: boolean; added: boolean };` parameter in the `FuncDiff` interface. Let me know if that looks good to you.

I think this is good. Did you check that it saves? I think it should automatically as it should detect the body of the function has changed

gidonkatten commented:
Merge conflicts need to be fixed btw

KacperKazan reviewed (approved):
great job Lancelot!! Keep it up!!!

lancelotblanchard commented:
> great job Lancelot!! Keep it up!!!

Thank you.
